frontend/packages/knative-plugin/src/components/add/SecretKeySelector.tsx

```diff
@@ -29,7 +29,13 @@ const SecretKeySelector: React.FC<SecretKeySelectorProps & StateProps> = ({
   const [secrets, setSecrets] = React.useState({});
   const fieldId = getFieldId(name, 'secret-key-input');
   const isValid = !(touched && error);
-  const errorMessage = !isValid ? error : '';
+  const errObj = (error as unknown) as { [key: string]: string };
+  const errMsg = `${errObj?.name ? 'Name is required ' : ''}${
```

Comment:
This could be modified into a generic error message creator function which takes an object of errors and returns a string. It could use the `key` and `value` of errors to create the error message properly without having the knowledge of what that key is going to be.